src/v1.js

```diff
@@ -35,16 +36,39 @@ api.declare({
   route:      '/persona',
   name:       'credentialsFromPersonaAssertion',
   idempotent: false,
+  input:      'persona-request.json',
   output:     'credentials-response.json',
   title:      'Get TaskCluster credentials given a Persona assertion',
   stability:  API.stability.experimental,
   description: [
     "Given an [assertion](https://developer.mozilla.org/en-US/Persona/" +
-    "Quick_setup#Step_4_Verify_the_user%E2%80%99s_credentials), return",
-    "an appropriate set of temporary credentials.",
+    "Quick_setup), return an appropriate set of temporary credentials.",
+    "",
+    "The supplied audience must be on a whitelist of TaskCluster-related",
+    "sites configured in the login service.  This is not a general-purpose",
+    "assertion-verification service!",
   ].join('\n')
 }, async function(req, res) {
-  return res.status(501).json({error: "not implemented yet"});
+  // verify the assertion with the persona service
+  let email;
+  try {
+    email = await this.personaVerifier.verify(req.body.assertion, req.body.audience);
+  } catch(err) {
+    // translate PersonaErrors into 400's; everything else is a 500
+    if (err.code == "PersonaError") {
+      return res.status(400).json(err);
```

Comment:
Use res.reportError(code, message,details), see:
https://github.com/taskcluster/taskcluster-lib-api/blob/master/src/errors.js#L22

For codes see:
https://github.com/taskcluster/taskcluster-lib-api/blob/master/src/errors.js#L5-L15
You can add additional code specific to your service if strictly necessary.